### ISSUE TITLE
Only LED discovered

### ISSUE THREAD

**Issue by Zealand67** [siku2/hass-dingz#7]
I do have the same issue as Schmidi198.
The error log show a few errors:
Unexpected error fetching state data: 'target'
15:35:05 – (ERROR) dingz (custom integration) - message first occurred at 14:25:32 and shows up 752 times
failed to create Blind from JSON: {'moving': 'stop', 'position': 0, 'lamella': 0, 'readonly': False, 'index': {'relative': 0, 'absolute': 1}}
15:35:05 – (ERROR) dingz (custom integration) - message first occurred at 14:25:32 and shows up 752 times
Can't connect to ESPHome API for 192.168.0.107 @ 192.168.0.107: Timeout while connecting to ('192.168.0.107', 6053)
15:27:52 – (WARNING) /usr/local/lib/python3.9/site-packages/aioesphomeapi/reconnect_logic.py
Empty reply found when expecting JSON data
15:05:19 – (WARNING) RESTful
Error fetching data: http://192.168.0.151/api/v1/state failed with
15:05:19 – (ERROR) RESTful
Update of sensor.dingz_test_state is taking over 10 seconds
15:05:18 – (WARNING) helpers/entity.py

I did a few reboots, reloads but still only have the LED integrated. Also this one shows status "unavailable"

Integration is version 1.5, HA 2021.11.5, Firmware on DINGZ is 1.3.30

Thanks for looking at it.

Some more Logs after reload of the integration:

**1st error**

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/dingz/sensor.py", line 22, in async_setup_entry
    if state.sensors.input_state is not None:
AttributeError: 'NoneType' object has no attribute 'sensors'

**2nd error:**

This error originated from a custom integration.

Logger: custom_components.dingz.api
Source: custom_components/dingz/api.py:32
Integration: dingz
First occurred: 17:20:29 (917 occurrences)
Last logged: 18:56:57

failed to create `Blind` from JSON: {'moving': 'stop', 'position': 0, 'lamella': 0, 'readonly': False, 'index': {'relative': 0, 'absolute': 1}}

**3rd error**

This error originated from a custom integration.

Logger: custom_components.dingz
Source: custom_components/dingz/api.py:96
Integration: dingz
First occurred: 17:20:29 (917 occurrences)
Last logged: 18:56:57

Unexpected error fetching state data: 'target'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 187, in _async_refresh
    self.data = await self._async_update_data()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 147, in _async_update_data
    return await self.update_method()
  File "/config/custom_components/dingz/__init__.py", line 46, in update_state
    return await session.state()
  File "/config/custom_components/dingz/api.py", line 642, in state
    return State.from_json(raw)
  File "/config/custom_components/dingz/api.py", line 37, in from_json
    raise e from None
  File "/config/custom_components/dingz/api.py", line 29, in from_json
    return cls._from_json(data.copy())
  File "/config/custom_components/dingz/api.py", line 272, in _from_json
    data["blinds"] = Blind.list_from_json(data["blinds"])
  File "/config/custom_components/dingz/api.py", line 41, in list_from_json
    return list(map(cls.from_json, data))
  File "/config/custom_components/dingz/api.py", line 37, in from_json
    raise e from None
  File "/config/custom_components/dingz/api.py", line 29, in from_json
    return cls._from_json(data.copy())
  File "/config/custom_components/dingz/api.py", line 96, in _from_json
    data["target"] = cls.Position.from_json(data["target"])
KeyError: 'target'

**Comment by siku2** [comment]
Thanks, that's very helpful. Seems like the API model for `Blind` is incorrect.
I don't really have the time to fix this right now, but if someone else is interested I'm willing to help out.

**Comment by epijim** [comment]
Does issue only occur if you have 2 blinds on one dingz? Like here and in #6 I get what I think is this issue, but only on the dingz where I have 2 blinds. On a dingz with 1 blind and 1-2 lights it works ok. 

The error is: 

```
Logger: homeassistant
Source: custom_components/dingz/light.py:102
Integration: dingz
First occurred: 9:04:26 PM (116 occurrences)
Last logged: 9:09:09 PM

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 182, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 330, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 135, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 390, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 566, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 607, in _async_write_ha_state
    state = self._stringify_state(available)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 572, in _stringify_state
    if (state := self.state) is None:
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1007, in state
    if (is_on := self.is_on) is None:
  File "/config/custom_components/dingz/light.py", line 118, in is_on
    return self._dingz_dimmer.on
  File "/config/custom_components/dingz/light.py", line 102, in _dingz_dimmer
    return self._dingz_state.dimmers[self._index]
IndexError: list index out of range
```

**Comment by OdyX** [comment]
This should be fixed in 0.1.6, and quite likely too in 2.0.0 (although with other problems).

**Comment by siku2** [comment]
Closing because it's probably no longer relevant.
